### PR TITLE
Add RHEL 8 support to Ansible tasks

### PR DESCRIPTION
**Pull request by ionutbalutoiu**
This update includes the following changes:

* Properly abstract the Ansible tasks to add new platform support
* Add RHEL 8 support
* Add more validation with GCC version check

### CODE REVIEW

**Comment by BRMcLaren** [comment]
@ionutbalutoiu Glad to have you contributing again!

We changed our license terms recently if you check out the DCO bot it will require a signoff and give steps to do so.

**Comment by ionutbalutoiu** [comment]
> @ionutbalutoiu Glad to have you contributing again!
> 
> We changed our license terms recently if you check out the DCO bot it will require a signoff and give steps to do so.

Thanks! PR is still work in progress. I'll make sure I include the signoff when I update the PR.

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
@BRMcLaren @paulcallen

This pull request is good to be reviewed and merged. I finished testing it.

It adds RHEL 8 support to Ansible tasks, and I took good care to maintain the backwards compatibility.

Merging this into `master` doesn't break anything since we still have to add the RHEL 8 testing stages to Bors (that is the next step).

**Comment by BRMcLaren** [comment]
Hello @ionutbalutoiu 

I signed off on your PR, I would like to ask that you check that [this](https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/OpenEnclave-E2E-Testing/detail/OpenEnclave-E2E-Testing/83/pipeline) completes before merging. 

There is nothing of concern and I believe this only adds desired functionality, but need to be sure when modifying dependencies as it can be scary to debug.

Thanks,
Brett

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
> Hello @ionutbalutoiu
> 
> I signed off on your PR, I would like to ask that you check that [this](https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/OpenEnclave-E2E-Testing/detail/OpenEnclave-E2E-Testing/83/pipeline) completes before merging.
> 
> There is nothing of concern and I believe this only adds desired functionality, but need to be sure when modifying dependencies as it can be scary to debug. I am sure you ran this regardless, but let's just make sure it is linked in the PR just in case.

The [OpenEnclave-E2E-Testing#83](https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/OpenEnclave-E2E-Testing/detail/OpenEnclave-E2E-Testing/83/pipeline) linked by you is green.

I just rebased against current master and Bors looks to be happy as well.

From my point of view, this is good to go.

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
bors cancel

**Comment by ionutbalutoiu** [comment]
bors try-

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
bors try-

**Comment by ionutbalutoiu** [comment]
bors try

**Comment by ionutbalutoiu** [comment]
Ignore above responses from our Bors. I just tested some infrastructure changes. Triggering it once again, and I'll let it go through.

bors try

**Comment by ionutbalutoiu** [comment]
bors try

File: scripts/ansible/roles/linux/docker/vars/redhat/oopta.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) Open Enclave SDK contributors.
```
Comment (oprinmarius) [inline scripts/ansible/roles/linux/docker/vars/redhat/oopta.yml]: I think here is a mistype and should be named ootpa.yml

Comment (ionutbalutoiu) [reply scripts/ansible/roles/linux/docker/vars/redhat/oopta.yml]: Oh, absolutely! Good catch, thanks!

**Comment by ionutbalutoiu** [comment]
bors try

**Pull request merged**